plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployCommandHandler.java

```diff
@@ -88,8 +86,10 @@ private IPath createWorkDirectory() throws IOException {
     return workDirectory;
   }
 
-  private Credential login(IShellProvider shellProvider) throws IOException, CoreException {
-    Credential credential = new GoogleLoginService().getActiveCredential(shellProvider);
+  private Credential login() throws CoreException {
+    IGoogleLoginService loginService =
+        PlatformUI.getWorkbench().getService(IGoogleLoginService.class);
```

Comment:
We want to avoid using `PlatformUI.getWorkbench()` wherever possible.  In this case, we're provided the `ExecutionEvent` which provides access to a bunch of objects that implement `IServiceLocator`.  Here's what I'd do:
  - pass in the `IGoogleLoginService` instance into `login()`
  - create a `getService(ExecutionEvent,Class<?>)` method that calls the following sequence of methods to find an `IServiceLocator`:
     - `HandlerUtil.getActiveSite()` (incl views and editors)
     - `HandlerUtil.getActiveEditor()`
     - `HandlerUtil.getActiveWorkbenchWindow()`
     - and if none of them work, then `PlatformUI.getWorkbench()`